src/utils/form-helper.test.ts

```diff
@@ -80,4 +95,404 @@ describe('Form Engine Helper', () => {
+      expectedDate.setHours(0, 0, 0, 0);
+
+      expect(parsedDate).toEqual(expectedDate);
+    });
+
+    it('should handle invalid date string format gracefully', () => {
+      const dateString = 'invalid-date-string';
+      const parsedDate = parseToLocalDateTime(dateString);
+
+      // Check if the parsedDate is an Invalid Date
+      expect(isNaN(parsedDate.getTime())).toBe(true);
+    });
+  });
+
+  describe('evaluateDisabled', () => {
+    let mockExpressionRunnerFn;
+    let node;
+    let allFields;
+    let allValues;
+    let sessionMode;
+    let patient;
+
+    beforeEach(() => {
+      mockExpressionRunnerFn = jest.fn();
+      node = { value: { disabled: { disableWhenExpression: '' } } };
+      allFields = [{ id: 'field1', value: 'value1' }];
+      allValues = { field1: 'value1' };
+      sessionMode = 'edit';
+      patient = { id: 'patient1', name: 'John Doe' };
+    });
+
+    test('returns true when disableWhenExpression evaluates to true', () => {
+      mockExpressionRunnerFn.mockReturnValue(true);
+      const result = evaluateDisabled(node, allFields, allValues, sessionMode, patient, mockExpressionRunnerFn);
+      expect(result).toBe(true);
+      expect(mockExpressionRunnerFn).toHaveBeenCalledWith(
+        node.value.disabled.disableWhenExpression,
+        node,
+        allFields,
+        allValues,
+        { mode: sessionMode, patient },
+      );
+    });
+
+    test('returns false when disableWhenExpression evaluates to false', () => {
+      mockExpressionRunnerFn.mockReturnValue(false);
+      const result = evaluateDisabled(node, allFields, allValues, sessionMode, patient, mockExpressionRunnerFn);
+      expect(result).toBe(false);
+      expect(mockExpressionRunnerFn).toHaveBeenCalledWith(
+        node.value.disabled.disableWhenExpression,
+        node,
+        allFields,
+        allValues,
+        { mode: sessionMode, patient },
+      );
+    });
+
+    test('works with different sessionMode values', () => {
+      sessionMode = 'view';
+      mockExpressionRunnerFn.mockReturnValue(true);
+      const result = evaluateDisabled(node, allFields, allValues, sessionMode, patient, mockExpressionRunnerFn);
+      expect(result).toBe(true);
+      expect(mockExpressionRunnerFn).toHaveBeenCalledWith(
+        node.value.disabled.disableWhenExpression,
+        node,
+        allFields,
+        allValues,
+        { mode: sessionMode, patient },
+      );
+    });
+
+    test('works with different patient data', () => {
+      patient = { id: 'patient2', name: 'Jane Doe' };
+      mockExpressionRunnerFn.mockReturnValue(false);
+      const result = evaluateDisabled(node, allFields, allValues, sessionMode, patient, mockExpressionRunnerFn);
+      expect(result).toBe(false);
+      expect(mockExpressionRunnerFn).toHaveBeenCalledWith(
+        node.value.disabled.disableWhenExpression,
+        node,
+        allFields,
+        allValues,
+        { mode: sessionMode, patient },
+      );
+    });
```

Comment:
I cannot imagine what could cause this test to fail

```suggestion
```

Reply:
Deleted